### ISSUE TITLE
Improve ByteBuffer/ByteBufferView equability

### ISSUE THREAD

**Issue by Davidde94** [apple/swift-nio#1870]
There are some instances, particularly in tests, where it would be really useful to provide a clean API for comparing a `ByteBuffer` and an expected array of bytes.

Current method 
```swift
XCTAssert(buffer.readableBytesView == ByteBufferView([0x05, 0x00]))
XCTAssert(buffer == ByteBuffer(bytes: [0x05, 0x00]))
```

I think we should provide a shorthand convenience, e.g.
```swift
XCTAssert(buffer == [0x05, 0x00])
XCTAssert(buffer.readableBytesView == [0x05, 0x00])
```

**Comment by Lukasa** [comment]
In this instance, I think it would be substantially more useful to conform `ByteBufferView` to `ExpressibleByArrayLiteral`, which would almost completely solve the problem.

**Comment by Patil2099** [comment]
@Davidde94  I would love to take this up.

**Comment by Davidde94** [comment]
@Patil2099 go for it! Reach out if you need any assistance

**Comment by mcomisso** [comment]
Hi, I wanted to give a look around and find a easy first issue, and noticed this one. 
I don't want to steal anybody's task or time as this was already assigned, but I've added this change on a feature branch in my swift-nio fork to play with the project.

@Patil2099 please let me know if you are still working on it and/or want to compare solutions, the branch is [this](https://github.com/mcomisso/swift-nio/tree/feature/ByteBufferView-ExpressibleByArrayLiteral).

Thanks!

**Comment by Davidde94** [comment]
@Patil2099 @mcomisso any progress? We'll happily take a look at PRs :)

**Comment by mcomisso** [comment]
I've opened a very small PR with the changes, hope you don't mind @Patil2099!

**Comment by Lukasa** [comment]
Resolved by #1886.
